casadm/cas_lib.c

```diff
@@ -820,10 +820,19 @@ struct cache_device *get_cache_device(const struct kcas_cache_info *info, bool b
 	cache->expected_core_count = info->info.core_count;
 	cache->id = cache_id;
 	cache->state = info->info.state;
-	if (set_device_path(cache->device, sizeof(cache->device), info->cache_path_name,
-			    sizeof(info->cache_path_name)) != SUCCESS) {
-		free(cache);
-		return NULL;
+	if (by_id_path) {
+		if (set_device_path(cache->device, sizeof(cache->device), info->cache_path_name,
```

Comment:
I suggest to simply copy info->cache_path_name. We expect it to be "by-id".

Reply:
Yeah, You're right, this works and simplifies function.